django/contrib/contenttypes/models.py

```diff
@@ -149,7 +149,7 @@ def name(self):
         model = self.model_class()
         if not model:
             return self.model
-        return str(model._meta.verbose_name)
+        return "{0} :: {1}".format(model._meta.app_label, model._meta.verbose_name)
```

Comment:
In keeping with the pattern that Permissions uses, I actually wanted to just use the `app_label`. It is guaranteed to be unique, and that's very important. For the model name, using the verbose name is also the pattern that Permissions uses. I think it makes more sense to do that in the model because they will already be namespaced by the app, and I think this is likely what was going through the mind of the developer that implemented this in Permissions. Perhaps another ticket could address the style of how the `app :: model` representation is done but then changing it in both Permissions and in ContentTypes would make sense.

Reply:
In other words, I'm not hostile to that change necessarily, but I think the first step is to get ContentType to show things in the same convention that is used in permissions. Then, additional improvements might make sense but I think those should be a separate issue.